Fixes #24143 - Show message when no org is selected on subs...

johnpmitsch opened the pull request:
page and RH repo page.

When no organization is selected, there is no context for RH repo
page and subs page and that page can't function. This commit
adds a message when you navigate to the page under "Any Organization"
to select an Organization.

We have also been sending requests when navigating to a page
without org context. This commit adds an org id check on applicable
actions. The request isn't sent if no organization ID is present.

This commit also adds a `common` folder in `webpack/` with a
`helpers` file. This is a  place for things that are katello
specific, but need to be shared across components. I'm open to
suggestions on how to structure this better.

johnpmitsch commented:
still cleaning up and adding tests, but feel free to review the functionality and overall approach

JacobCallahan commented on webpack/components/OrganizationCheck/index.js:
```diff
@@ -0,0 +1,40 @@
+            <div class="centered-message">
+              <h2>No organization selected</h2>
+              <h3>
+                The page you are attempting to access requires selecting a specific organization.
```
The way this reads is that there is only one correct Organization to choose. Can you change the text to something like:
"...requires a selected organization"

johnpmitsch commented:
Screenshot of RH repos page:

![h](http://www.clipular.com/c/4779730558255104.png?k=jbf3lbkMvcbpcVoPV5-FW9oaydM)

johnpmitsch commented:
Updated to remove WIP so this is ready for review, I've added tests and linted. For some reason I had to fix linting issues in code I didn't touch, I'm not sure what the reason for that is

johnpmitsch commented:
@stbenjam +1 to keeping things consistent as the angular page (with the org drop down) was written before the top menu bar with org selector was added, so seems reasonable to update.

@Rohoover does this sound ok to you? To recap, we want to  change the old org selector page to look more like the new one written in React, removing the org selector in the page. (see @stbenjam screenshots) This is based on the premise that the dropdown org selector is already in the top menu bar and we shouldn't display an additional dropdown.

waldenraines commented:
@amirfefer how does this compare with what you're working on for http://projects.theforeman.org/issues/22568?

waldenraines commented on webpack/common/helpers.js:
```diff
@@ -0,0 +1,16 @@
+const handleMissingOrg = (orgId, dispatch, type) => {
+  if (!orgId) {
+    const errorMessage = 'No organization specified';
```
This will need to be translated with `__()`.

waldenraines commented on webpack/components/OrganizationCheck/__tests__/OrganizationCheck.test.js:
```diff
@@ -0,0 +1,24 @@
+import OrganizationCheck from '../../OrganizationCheck';
+
+describe('Organization check page', () => {
+  /* eslint-disable react/jsx-closing-tag-location */
```
I've gotten around this before by setting the element to a `const`.  We may want to revisit this rule in the future.

waldenraines commented on webpack/components/OrganizationCheck/index.js:
```diff
@@ -0,0 +1,41 @@
+              </h3>
+              <h3>
+                Please select an organization from the dropdown in the top menu bar
+                and you will be redirected.
```
All of this needs to be translated with `__()`.

waldenraines commented on webpack/components/OrganizationCheck/index.js:
```diff
@@ -0,0 +1,41 @@
+      <Grid bsClass="container-fluid">
+        <Row>
+          <Col sm={12}>
+            <div className="centered-message">
```
Could you just use [`.text-center`](https://getbootstrap.com/docs/3.3/css/#type-alignment) here?

johnpmitsch commented:
@waldenraines what is your reasoning for adding another organization dropdown in the page? There already is one visible in the top menu bar.

waldenraines commented:
> @waldenraines what is your reasoning for adding another organization dropdown in the page? There already is one visible in the top menu bar.

@johnpmitsch consistency with ng-katello and because we may as well make how to select the organiozation obvious since this effectively blocks the user from doing anything.  Basically the same reason we added it in the first place.  I will, however, defer to @rohoover on whether or not we should include it.  If we don't include it here we should also remove it from the katello angular pages.

johnpmitsch commented:
We will most likely use https://github.com/Katello/katello/pull/7495 instead of this PR so hold off on reviews here for now!

Rohoover commented:
@johnpmitsch @waldenraines 

I see this as an empty state for the user. The current empty state patterns offer messaging and the ability to do an action. More importantly, we should be consistent how we're handling this between the different efforts as @waldenraines mentions.

johnpmitsch commented:
@Rohoover thanks for the response, I'm going to close this in favor of https://github.com/Katello/katello/pull/7495  since that seems to be in line with what you are suggesting
